src/construction/construction_read_validate.py

```diff
@@ -54,72 +50,85 @@ def read_validate_construction_files(
-            file_exists_func=check_file_exists,
-        )
-        if isinstance(pc_construction_df, type(None)):
-            run_postcode_construction = False
-            pc_construction_df = pd.DataFrame()
     else:
-        pc_construction_df = pd.DataFrame()
-
-    if run_postcode_construction:
-        validate_data_with_schema(pc_construction_df, postcode_schema_path)
+        # Validate construction data and check it doesn't contain duplicates
+        validate_data_with_schema(construction_df, schema_path)
         check_for_duplicates(
-            df=pc_construction_df,
+            df=construction_df,
             columns=["reference", "instance"],
             logger=construction_logger,
         )
-        validate_columns_not_empty(
-            df=pc_construction_df,
-            columns=["601", "referencepostcode"],
-            logger=construction_logger,
-            _raise=True,
-        )
-        pc_construction_df["force_imputation"] = True
 
-    return construction_df, pc_construction_df
+    return construction_df
+
+
+def read_validate_postcode_construction_file(
+    config: dict,
+    check_file_exists: Callable,
+    read_csv: Callable,
+    construction_logger
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Read and validate construction files for the construction module.
+
+    Function reads config to determine which constructions are required/if it
+    is for Northern Ireland. It then reads the construction files and validates.
+
+    Args:
+        config (dict): The pipeline configuration.
+        check_file_exists (Callable): Function to check if file exists.
+        read_csv (Callable): Function to read a csv file.nstruction (bool, optional): If true, run the postcode
+            only construction.
+
+    Returns:
+        pd.DataFrame: The construction dataframes for postcode constructions.
+    """
+
+    # read in postcode construction file
+    paths = config["construction_paths"]
+    postcode_construction_fpath = paths["postcode_construction_file_path"]
+    postcode_schema_path = "./config/postcode_construction_schema.toml"
+
+    if not check_file_exists(postcode_construction_fpath):
+        raise FileNotFoundError(
+            f"PC construction file not found at: {postcode_construction_fpath}"
+        )
+    pc_construction_df = read_construction_file(
+        path=postcode_construction_fpath,
+        logger=construction_logger,
+        read_csv_func=read_csv,
+        file_exists_func=check_file_exists,
+    )
+
+    validate_data_with_schema(pc_construction_df, postcode_schema_path)
+    check_for_duplicates(
+        df=pc_construction_df,
+        columns=["reference", "instance"],
+        logger=construction_logger,
+    )
+    validate_columns_not_empty(
+        df=pc_construction_df,
+        columns=["601", "referencepostcode"],
+        logger=construction_logger,
+        _raise=True,
+    )
+    pc_construction_df["force_imputation"] = True
```

Comment:
We don't actually need this now. To think about, is there a way to identify if a record has gone through postcode construction?

Reply:
Good point - I'll take that out.  Postcode constructions are flagged in the is_constructed col